### PR TITLE
[DSME] Use wheel group as priveleged

### PR DESCRIPTION
**Pull request by neochapay**
In nemo we don't have privileged user but we can use wheel group for detect privileged users

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -168,6 +168,18 @@ AS_IF([test "x$enable_abootsettings" != xno],
 AM_CONDITIONAL([WANT_ABOOTSETTINGS], [test x$enable_abootsettings != xno])
 
 
+#
+# Wheell is priveleged
```
Comment (spiiroin) [inline configure.ac]: typos: '"Wheell', "priveleged"

("priveleged" appears in couple other places too)

File: dsme/utility.c [hunk 2]
```diff
@@ -64,6 +66,24 @@ dsme_user_is_privileged(uid_t uid, gid_t gid)
...
+        int ngroups = 0;
+        getgrouplist(pw->pw_name, pw->pw_gid, NULL, &ngroups);
+        gid_t groups[ngroups];
+        getgrouplist(pw->pw_name, pw->pw_gid, groups, &ngroups);
+
+        for (int i = 0; i < ngroups; i++) {
+            struct group* gr = getgrgid(groups[i]);
+            if( gr != NULL ) {
+                if(strcmp(gr->gr_name,"wheel") != 0 ) {
+                   is_privileged = true;
```
Comment (spiiroin) [inline dsme/utility.c:81]: All groups except "wheel" are privileged? Should be `if( strcmp() == 0 )`?

Also, once a match is found, one might as well break out of the loop.

File: configure.ac [hunk 3]
```diff
@@ -168,6 +168,18 @@ AS_IF([test "x$enable_abootsettings" != xno],
...
+#
+AC_ARG_ENABLE([usewheel],
+  [AS_HELP_STRING([--enable-usewheel],
+    [use wheel group users as priveleged])])
```
Comment (spiiroin) [inline configure.ac]: I'm wondering if that explanation could be clarified a bit?

Say, something like "Users in wheel group are considered privileged", or maybe "Only users..." depending on whether root user that does not belong to wheel group is intentionally excluded or not.

**Review by spiiroin — CHANGES_REQUESTED** [review]

File: configure.ac [hunk 4]
```diff
@@ -168,6 +168,18 @@ AS_IF([test "x$enable_abootsettings" != xno],
...
+# Wheel is privileged
+#
+AC_ARG_ENABLE([usewheel],
+  [AS_HELP_STRING([--enable-usewheel],
+    [Users in wheel group are considered privileged])])
+
+AS_IF([test "x$enable-usewheel" != xno],
+  [AC_DEFINE([DSME_USEWHEEL], [1], [Users in wheel group are considered privileged])])
+
+AM_CONDITIONAL([WANT_USEWHEEL], [test x$enable-usewheel != xno])
```
Comment (spiiroin) [inline configure.ac]: Looks like DSME_USEWHEEL ends up always defined regardless of configure options - which is not ok.

After taking a brief AC_ARG_ENABLE refresher... AFAICT enabling "disabled-by-default" config item (like this usewheel here) should use "x$topic" == xyes" type tests.

(On a quick glance a lot of the other already existing config options have similar issues, but fixing those falls outside the context of this pull req)

**Review by spiiroin — CHANGES_REQUESTED** [review]

**Comment by PureTryOut** [comment]
Depending on the distro, it might not actually be "wheel" that has "sudo rights". I've seen other group names before, I think even in plain Debian (but I can't recall right now). Would it maybe be possible to parse `/etc/sudoers` to verify which group is privileged (if any at all)?  Then again, you maybe don't want to depend on `sudo` and `/etc/sudoers` being available, people could want or have `doas` instead. Is there not some other way to check if the user has the required rights? Or maybe just make the group name configurable at compile time (and default to `wheel`), that's probably the easiest.

**Comment by spiiroin** [comment]
Ad-hoc test: building sans --enable-usewheel -> uses root/privileged as before (= what sailfish needs); building with  --enable-usewheel -> only users added to the wheel group are privileged = matches what I understand the intent of this pr to be -> LGTM

> Or maybe just make the group name configurable at compile time (and default to `wheel`), that's probably the easiest.

Assuming that: 1) there is concrete need for wheel group use (= a problem that this pr solves), 2) apparently at the moment there is no known and well defined need for more degrees of freedom, and 3) making the group name configurable can be done later on too -> I would not place burden of potentially unnecessary future proofing on this pull req... So I'd leave it up to: @neochapay, do you want to implement also such group name config option? Or should we just merge this as it is now?

**Comment by neochapay** [comment]
@spiiroin merge as is. If @PureTryOut need some other thinks it make own MR :)

**Comment by PureTryOut** [comment]
I don't need anything personally, my system (postmarketOS/Alpine Linux) also uses the wheel group so I'm fine there. I was just thinking further and "what if other systems would use this, what problems might they encounter"? Trying to improve the whole thing  for everyone basically, and just thinking out loud :wink:

**Review by spiiroin — APPROVED** [review]

**Comment by spiiroin** [comment]
Included in dsme 0.84.0 tag

**Pull request merged**